test/built-ins/Function/internals/Call/class-ctor-realm.js

```diff
@@ -3,17 +3,20 @@
 /*---
 esid: sec-ecmascript-function-objects-call-thisargument-argumentslist
 description: >
-  Error when invoking a class constructor (honoring the Realm of the current
-  execution context)
-info: |
-  [...]
-  2. If F's [[FunctionKind]] internal slot is "classConstructor", throw a
-     TypeError exception.
+  Error when invoking a default class constructor, honoring the Realm
+  that the class was defined in.
 features: [cross-realm, class]
 ---*/
 
-var C = $262.createRealm().global.eval('0, class {}');
+const realm = $262.createRealm();
+const C = realm.global.eval('(class {})');
+const TE = realm.global.eval('TypeError');
```

Comment:
Can you walk me through this? I'm not sure I understand how the TypeError thrown in this Realm would come from another Realm (thanks in advance)

Reply:
What TypeError is thrown in the main realm?

Reply:
The try/catch below is in the main Realm, so the object it creates will be from the main Realm.

```js
function otherRealm() {
  let frame = document.createElement('iframe');
  document.body.appendChild(frame);
  frame.contentDocument.write('<body>');
  const global = frame.contentWindow;
  return {
    global
  };
}


const other = otherRealm();
const C = other.global.eval('(class {})');

try {
  C();
} catch (error) {
  console.log(error instanceof globalThis.TypeError); // true
  console.log(error instanceof other.global.TypeError); // false
}
```

Or in V8, if you prefer: 

```js
let realmId = Realm.createAllowCrossRealmAccess();
let other = Realm.global(realmId);
const C = other.eval('(class {})');

try {
  C();
} catch (error) {
  print(error instanceof globalThis.TypeError); // true
  print(error instanceof other.TypeError); // false
}
```

Reply:
right those engines implement the spec as it is currently written, with my pr the error will come from the other realm.

Reply:
That's what I missing. Thanks.